docs/docs/guides/18_migration_viem/index.md

```diff
@@ -4,4 +4,261 @@ title: 'Migration from Web3.js to Viem'
 position: 18
 ---
 
-Coming soon.
+# Migration from Web3.js to Viem
+
+This guide will help you migrate from Web3.js v4 to Viem v2 for interacting with the Ethereum blockchain. The guide provides code examples for both libraries for transition.
+
+## Installation
+
+To begin migrating from Web3.js to Viem, first install the Viem package:
+
+```bash
+npm install viem@2
+```
+
+## Providers
+
+When migrating from Web3.js to Viem, the first step is to update how you connect to the Ethereum network. Both libraries use providers, but their initialization differs.
+
+```javascript
+import Web3 from 'web3';
+
+const web3 = new Web3(providerURL);
+```
+
+To migrate this to Viem, you'll need to replace it with using `createPublicClient()`. This function creates a client for interacting with the Ethereum network.
+
+```javascript
+import { createPublicClient, http } from 'viem';
+
+const client = createPublicClient({ transport: http(providerURL) });
+const blockNumber = await client.getBlockNumber();
+```
+
+## Browser-injected Provider
+
+For browser wallet connections like MetaMask, update how you handle the injected provider.
+
+```javascript
+const web3 = new Web3(window.ethereum);
+```
+
+To migrate this to Viem, you'll need to use `createWalletClient()` with `custom()` instead of creating a new Web3 instance.
+
+```javascript
+import { createWalletClient, custom } from 'viem'
+import { mainnet } from 'viem/chains'
+
+const client = createWalletClient({
+  chain: mainnet,
+  transport: custom(window.ethereum!)
```

Comment:
nit: inconsistency, window.ethereum with exclamation mark in viem and without in web3js snippets.